src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md

```diff
@@ -1,4 +1,400 @@
+[GitHub](https://github.com/topics/xk6) and wish to use it in your tests. Great! The
+process is relatively simple and we're working on simplifying it further for end
+users.
+
+You'll first need to setup [Go](https://golang.org/doc/install) and
+[Git](https://git-scm.com/). Make sure that your `$PATH` environment variable is
+updated and that `go version` returns the correct version.
+
+Then install xk6 with:
+
+<CodeGroup labels={[]} lineNumbers={[false]}>
+
+```bash
+$ go install github.com/k6io/xk6/cmd/xk6@latest
+```
+
+</CodeGroup>
+
+And confirm that `which xk6` on Linux/macOS or `where xk6` on Windows returns a
+valid path. Otherwise ensure that `$GOPATH` is correctly defined and that
+`$GOPATH/bin` is added to your `$PATH` environment variable. See the
+[Go documentation](https://golang.org/cmd/go/#hdr-GOPATH_environment_variable) for details.
+
+Once xk6 is installed, building a k6 binary with one or more extensions can be done
+with the following command:
+
+<CodeGroup labels={[]} lineNumbers={[false]}>
+
+```bash
+$ xk6 build v0.33.0 \
+  --with github.com/imiric/xk6-sql \
+  --with github.com/szkiba/xk6-prometheus
+```
+
+</CodeGroup>
+
+This will build a `k6` binary in the current directory based on k6 v0.33.0, bundling
+a JavaScript and an Output extension. Now you can run a script with this binary that
+uses the [`xk6-sql` JS API](https://github.com/imiric/xk6-sql) and the
+[Prometheus output](https://github.com/szkiba/xk6-prometheus).
+
+Note that when running the script we have to specify the binary just built in the
+current directory (`./k6`), as otherwise some other `k6` binary found on the system
+could be executed which might not have the extensions built-in. This is only the case
+on Linux and macOS, as Windows shells will execute the binary in the current
+directory first.
+
+
+## Writing a new extension
+
+The first thing you should do before starting work on a new extension is to confirm
+that a similar extension doesn't already exist for your use case. Take a look at
+the [Ecosystem page](/ecosystem) and the [`xk6` topic on GitHub](https://github.com/topics/xk6).
+For example, if a system you need support for can be tested with a generic protocol
+like MQTT, prefer using [xk6-mqtt](https://github.com/pmalhaire/xk6-mqtt)
+instead of creating an extension that uses some custom protocol.
+
+Next, you should decide the type of extension you need. A JavaScript extension is a
+good fit if you want to extend the JS functionality of your script, or add support
+for a new network protocol to test with. An Output extension would be more suitable
+if you need to process the metrics emitted by k6 in some way, submit them to a
+specific storage backend that was previously unsupported, etc. The k6 APIs you'll
+need to use and things to consider while developing will be different in each case.
+
+
+### Writing a new JavaScript extension
+
+A good starting point for using xk6 and writing a JS extension is the [xk6
```

Comment:
Given that most users will have problems with the context passing after some point I do think that this section should either have a disclaimer that this is an extremely simple example or to be expanded to show the "module v2" way and possibly a note about common.Bind and ... not using it ;)

Reply:
Hhmm yeah, that's tricky. On the one hand, it's going to be confusing if we have to document the `common.Bind()` way and the `ModuleV2` way... But on the other, suggesting `common.Bind()` would be outdated quickly and we probably don't want to suggest that approach moving forward, and I'm not sure if `ModuleV2` can be considered stable for public use yet.

@na-- WDYT?

Reply:
I am for just documenting `ModuleV2` with a warning (that should be either way there) that internal APIs can change and that the whole thing is alpha like. We can also mention `common.Bind`, link to the issue I made about it and say that it's *not* advisable to use it at all and moving to `ModuleV2` is the better plan. 

I doubt we will just remove `common.Bind` in 2-6 months and likely `ModuleV2` will just get a lot more methods. The only one (unfortunately) that I don't like is `GetRuntime` and mostly due to the whole fact that it lets you have the Runtime indefinitely even if (in the future) we have something asynchronous going on as well. But at least I don't have any non-trivial and likely non-performant ideas for it, so we likely will have to keep it and have a long explanation on how ... you might want to be careful with it, one day.